### PR TITLE
docs: update doc install_deploy.md of zookeeper 3.4.14 java requirements

### LINKED ISSUE #3766: zookeeper-3.4.14 client throw "Unable to canonicalize address localhost/<unresolved>:7181 because it's not resolvable" when using jdk 18.0.2.1
**Bug Description**
jdk: 
```
java version "18.0.2.1" 2022-08-18
Java(TM) SE Runtime Environment (build 18.0.2.1+1-1)
Java HotSpot(TM) 64-Bit Server VM (build 18.0.2.1+1-1, mixed mode, sharing)
```

zookeeper config:
```
# The number of milliseconds of each tick
tickTime=2000
# The number of ticks that the initial 
# synchronization phase can take
initLimit=10
# The number of ticks that can pass between
# sending a request and getting an acknowledgement
syncLimit=5
# the directory where the snapshot is stored.
# do not use /tmp for storage, /tmp here is just
# example sakes.
dataDir=./data
# the port at which the clients will connect
clientPort=7181
# the maximum number of client connections.
# increase this if you need to handle more clients
#maxClientCnxns=60
#
# Be sure to read the maintenance section of the
# administrator guide before turning on autopurge.
#
# http://zookeeper.apache.org/doc/current/zookeeperAdmin.html#sc_maintenance
#
# The number of snapshots to retain in dataDir
#autopurge.snapRetainCount=3
# Purge task interval in hours
# Set to "0" to disable auto purge feature
#autopurge.purgeInterval=1
```

cmd: `bash bin/zkCli.sh -server localhost:7181`

exceptions:
```log
2024-02-21 12:00:54,088 [myid:] - INFO  [main:ZooKeeper@442] - Initiating client connection, connectString=localhost:7181 sessionTimeout=30000 watcher=org.apache.zookeeper.ZooKeeperMain$MyWatcher@3c679bde
Welcome to ZooKeeper!
JLine support is enabled
[zk: localhost:7181(CONNECTING) 0] 2024-02-21 12:00:54,206 [myid:] - WARN  [main-SendThread(localhost:7181):ClientCnxn$SendThread@1164] - Session 0x0 for server localhost/<unresolved>:7181, unexpected error, closing socket connection and attempting reconnect
java.lang.IllegalArgumentException: Unable to canonicalize address localhost/<unresolved>:7181 because it's not resolvable
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:65)
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:41)
	at org.apache.zookeeper.ClientCnxn$SendThread.startConnect(ClientCnxn.java:1001)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1060)
2024-02-21 12:00:55,317 [myid:] - WARN  [main-SendThread(localhost:7181):ClientCnxn$SendThread@1164] - Session 0x0 for server localhost/<unresolved>:7181, unexpected error, closing socket connection and attempting reconnect
java.lang.IllegalArgumentException: Unable to canonicalize address localhost/<unresolved>:7181 because it's not resolvable
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:65)
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:41)
	at org.apache.zookeeper.ClientCnxn$SendThread.startConnect(ClientCnxn.java:1001)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1060)
```

**Expected Behavior**
update docs, clarify compatible jdk versions

**Relation Case**

**Steps to Reproduce**

1. `bash bin/zkServer.sh start`
2. `bash bin/zkCli.sh -server localhost:7181`

### PR DESCRIPTION
**Pull request by yangwucheng**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
docs

* **What is the current behavior?** (You can also link to an open issue here)
fix #3766

* **What is the new behavior (if this is a feature change)?**

### CODE REVIEW

File: docs/zh/deploy/install_deploy.md [hunk 1]
```diff
@@ -21,6 +21,8 @@ strings /lib64/libc.so.6 | grep ^GLIBC_
 
 如果需要部署 ZooKeeper 和 TaskManager，则需要有 Java 运行环境。
 
+> Zookeeper 3.4.14需要 `Java 1.7` - `Java 17` 版本。推荐使用 `Java 1.8` 或 `Java 11` 版本。
```
Comment (vagetablechicken) [inline docs/zh/deploy/install_deploy.md]: https://github.com/4paradigm/OpenMLDB/issues/3766#issuecomment-1955863808
jre 14 and 17failed too, so it should be 'zookeeper 3.4.14 can't work on jre >= 14', only zk 3.4.14 we provided. If users use their own higher zk, higher version jre is ok.

**Comment by aceforeverd** [comment]
As I remember, there probably some issues running java tests on JDK11, just not caused by zookeeper. So it may necessary to verify JDK11 support from development and usage perspective

**Comment by vagetablechicken** [comment]
> As I remember, there probably some issues running java tests on JDK11, just not caused by zookeeper. So it may necessary to verify JDK11 support from development and usage perspective 

4pdosc/openmldb is jdk 11, no zk issues?

**Comment by aceforeverd** [comment]
> 4pdosc/openmldb is jdk 11, no zk issues?

just `mvn test` in source code, probably fine from usage

**Comment by vagetablechicken** [comment]
> > 4pdosc/openmldb is jdk 11, no zk issues?
> 
> 
> 
> just `mvn test` in source code, probably fine from usage

We just want to fix usage doc here, dev doc should add 11 unsupported.

**Review by aceforeverd — APPROVED** [review]

**Review by tobegit3hub — APPROVED** [review]
LGTM

File: docs/zh/deploy/install_deploy.md [hunk 2]
```diff
@@ -21,6 +21,8 @@ strings /lib64/libc.so.6 | grep ^GLIBC_
 
 如果需要部署 ZooKeeper 和 TaskManager，则需要有 Java 运行环境。
```
Comment (vagetablechicken) [inline docs/zh/deploy/install_deploy.md:22]: server运行也给个版本，1.8及以上

File: docs/zh/deploy/install_deploy.md [hunk 3]
```diff
@@ -21,6 +21,8 @@ strings /lib64/libc.so.6 | grep ^GLIBC_
...
 
+> Zookeeper Client 3.4.14 需要 `Java 1.7` - `Java 13` 版本，如果希望使用zkCli，推荐使用 `Java 1.8` 或 `Java 11` 版本。
```
Comment (dl239) [inline docs/zh/deploy/install_deploy.md]: update en docs.  docs/en/deploy/install_deploy.md

**Review by Elliezza — APPROVED** [review]
添加对应英文改动

**Review by dl239 — APPROVED** [review]

**Pull request merged**